lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@rocksfrow I doubt that the server is checking the DSN option. I am pretty sure that it is the client that accepts or denies that option so working with the PHP constant would be the best option here. Catching an exception for BC on connect is a really bad idea as I think it causes more overhead than just performing `SET NAMES` (building the stack trace for exceptions is really expensive).